src/utils/PrivateRoute.tsx

```diff
@@ -0,0 +1,14 @@
+import type { FC, PropsWithChildren } from 'react';
+import { Navigate } from 'react-router-dom';
+
+import useLoggedInUser from '../hooks/useLoggedInUser';
+
+export const PrivateRoute: FC<PropsWithChildren> = ({ children }) => {
```

Comment:
@Antalyan netušíš, jak se tady zbavit té TS chyby?
```
Type '({ children }: { children?: ReactNode; }) => string | number | boolean | ReactFragment | Element | null | undefined' is not assignable to type 'FC<{ children?: ReactNode; }>'.
  Type 'string | number | boolean | ReactFragment | Element | null | undefined' is not assignable to type 'ReactElement<any, any> | null'.
    Type 'undefined' is not assignable to type 'ReactElement<any, any> | null'.ts(2322)
```

Reply:
Pushnu commit s opravou, překlopil jsem to do ternárního operátoru a obalil React fragmentem, bůhvíproč se to nelíbí eslintu, ale s tím se asi můžeme smířit.